lib/TokenManager.ts

```diff
@@ -192,6 +193,18 @@ function clear(tokenMgmtRef, storage) {
   storage.clearStorage();
 }
 
+function isTooManyRenews(tokenMgmtRef, renewTimeQueue) {
+  let tooManyRenews = false;
+  renewTimeQueue.push(Date.now());
```

Comment:
I would prefer that this function (which promises to answer a question true/false) could be called multiple times without side-effect and give the same answer. It could be broken into 3 functions (track, evaluate, cleanup), 

but it would be simpler to rename to something like 'allowRenewRequest' or `throttleRenew` with a comment explaining there is some time-based and stateful flood prevention logic here to prevent accidental DDoS and/or infinite loops